internal/controllers/machine/machine_controller.go

```diff
@@ -368,7 +368,7 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 		conditions.MarkTrue(m, clusterv1.PreDrainDeleteHookSucceededCondition)
 
 		// Drain node before deletion and issue a patch in order to make this operation visible to the users.
-		if r.isNodeDrainAllowed(m) {
+		if r.isNodeDrainAllowed(m) && !conditions.IsTrue(m, clusterv1.DrainingSucceededCondition) {
```

Comment:
Is this true? When might we need to perform a second drain?

Reply:
In my opinion we should not skip drain if it completed once. 

This would somehow allow to not drain again once all pods were gone, but this is only relevant in cases where Pods are rescheduled. But if one of these Pods has a volume we're then stuck in wait for volume detach